remove ProgressOpenEdgeDialect as base dialect (#673)
tableau/connector-plugin-sdk: docs/design.md

Choosing the right dialect is a critical part of writing a connector. For example:

> - If you set <span style="font-family: courier new">base</span> to "mysql_odbc" as the parent, you can skip configuring a dialect, and your connector will use the parent’s dialect.

Reviewer: Can you invert the ordering of these to better show our recommendations. Complete is ideal, followed by inherit base, followed by superclass

Author: Done in the latest commit, thanks, I just committed your suggestions for other comments, thanks for writing them out for me.

> - Inherit a base dialect scenario: If your database follows the SQL standards of another database that Tableau already supports (listed below), then you can choose that dialect as a starting point, you can also add your own customizations beyond the base dialect.

Reviewer 2: I'd add a warning, either here or in the "Should I create a dialect definition" section below (or preferably both), that a new version of Tableau may change the base dialects and cause unintended side effects for connectors inheriting from the base dialect. We want to discourage people from tying their connector to our connectors as much as possible.

Author: Done, thank you!

> - Write a complete dialect on your own scenario: you can create a complete dialect definition without any base dialect.

Reviewer 2: Maybe point to our full_tdd or the full postgres sample as a starting point?

Reviewer: Postgres samples are preferred to the component examples.

Author: Done, thank you!

> - Write a complete dialect on your own scenario: you can create a complete dialect definition without any base dialect. You can refer to [Postgres_odbc sample's full dialect](https://github.com/tableau/connector-plugin-sdk/blob/master/samples/plugins/postgres_odbc/dialect.tdd) as a starting point.

Reviewer: ```suggestion
- Write a complete dialect scenario (recommended): You can create a complete dialect definition without any base dialect. Refer to the [Postgres_odbc sample's full dialect](https://github.com/tableau/connector-plugin-sdk/blob/master/samples/plugins/postgres_odbc/dialect.tdd) as a starting point.
```

- Teradata1410Dialect
- VerticaDialect

> One warning to note is that a new version of Tableau may change those base dialects' behavior and may cause unintended side effects for connectors inheriting from those base dialects.

Reviewer: ```suggestion
```
I think it makes more sense merged with the statement below.

### Should I create a dialect definition file?

> If you want to customize the generated SQL instead of using the superclass's dialect, or if your connector inherits from ODBC or JDBC, then you need to create a custom dialect file. If your database follows the SQL standards of any database that Tableau currently supports(listed above), then you can choose that base dialect as a starting point.

Reviewer: ```suggestion
If you want to customize the generated SQL, or if your connector inherits from ODBC or JDBC, then you need to create a custom dialect file. 

Only if your database follows the SQL standards of a database that Tableau currently supports (listed above) should you consider choosing that base dialect as a starting point.  Also, any new Tableau version can change those base dialects' behavior, which may cause unintended side effects for connectors inheriting from those base dialects.
```
